PepperDashEssentials/Bridges/JoinMaps/SetTopBoxControllerJoinMap.cs

```diff
@@ -0,0 +1,158 @@
+        public uint FFwd { get; set; } //
+        public uint Pause { get; set; } //
+        public uint Play { get; set; } //
+        public uint Record { get; set; }
+        public uint Rewind { get; set; } //
+        public uint Stop { get; set; } //
+
+        public uint PowerOn { get; set; } //
+        public uint PowerOff { get; set; } //
+        public uint PowerToggle { get; set; } //
+
+        public uint HasKeypadAccessoryButton1 { get; set; }
+        public uint HasKeypadAccessoryButton2 { get; set; }
+
+        public uint KeypadAccessoryButton1Press { get; set; }
+        public uint KeypadAccessoryButton2Press { get; set; }
+
+
+        public uint HasDvr { get; set; }
+        public uint HasPresets { get; set; }
+        public uint HasNumeric { get; set; }
+        public uint HasDpad { get; set; }
+
+
+        #endregion
+
+        #region Analogs
+
+        #endregion
+
+        #region Strings
+        public uint Name { get; set; }
+        public uint LoadPresets { get; set; }
+        public uint KeypadAccessoryButton1Label { get; set; }
+        public uint KeypadAccessoryButton2Label { get; set; }
+
+        #endregion
+
+        public SetTopBoxControllerJoinMap()
+        {
+            PowerOn = 1;
+            PowerOff = 2;
+            PowerToggle = 3;
+
+            HasDpad = 4;
+            Up = 4;
+            Down = 5;
+            Left = 6;
+            Right = 7;
+            Select = 8;
+            Menu = 9;
+            Exit = 10;
+
+            HasNumeric = 11;
+            Digit0 = 11;
+            Digit1 = 12;
+            Digit2 = 13;
+            Digit3 = 14;
+            Digit4 = 15;
+            Digit5 = 16;
+            Digit6 = 17;
+            Digit7 = 18;
+            Digit8 = 19;
+            Digit9 = 20;
+            Dash = 21;
+            KeypadEnter = 22;
+            ChannelUp = 23;
+            ChannelDown = 24;
+            LastChannel = 25;
+
+            Guide = 26;
+            Info = 27;
+            Red = 28;
+            Green = 29;
+            Yellow = 30;
+            Blue = 31;
+
+            HasDvr = 32;
+            DvrList = 32;
+            Play = 33;
+            Pause = 34;
+            Stop = 35;
+            FFwd = 36;
+            Rewind = 37;
+            ChapPlus = 38;
+            ChapMinus = 39;
+            Replay = 40;
+            Record = 41;
+            HasKeypadAccessoryButton1 = 42;
+            KeypadAccessoryButton1Press = 42;
+            HasKeypadAccessoryButton2 = 43;
+            KeypadAccessoryButton2Press = 43;
+
+            Name = 1;
+            KeypadAccessoryButton1Label = 42;
+            KeypadAccessoryButton2Label = 43;
+
+            LoadPresets = 50;
+        }
+
+        public override void OffsetJoinNumbers(uint joinStart)
+        {
+            var joinOffset = joinStart - 1;
+            var properties = this.GetType().GetCType().GetProperties().Where(o => o.PropertyType == typeof(uint)).ToList();

```

Comment:
Please stop using reflection to offset join maps. It precludes someone from using custom join maps in their configuration to move things around if need be.